### PR TITLE
Feature/85 forgot password

### PR DESCRIPTION
**Pull request by fasbrothers**
Implemented:

- send email info to the backend
- receive a token from the email link and add a password to restore the user's password. Then send it back to the backend.

Additionally:

- update sign-in and sign-up with Zod, remove unnecessary codes

### CODE REVIEW

File: src/routes/routes.tsx [hunk 1]
```diff
@@ -9,10 +9,12 @@ import UserRoute from './UserRoute';
 import MainLayout from '@src/layouts/MainLayout';
 import UserLayout from '@src/layouts/UserLayout';
 
+import ResetPassword from '@src/pages/auth/ResetPassword';
+import SignIn from '@src/pages/auth/SignIn';
+import Signup from '@src/pages/auth/Signup';
```
Comment (Sinumi) [inline src/routes/routes.tsx]: Is there are reason for not using lazy for these routes?

File: src/components/input/PasswordInput.tsx
Comment (Sinumi) [inline src/components/input/PasswordInput.tsx:1]: nice

File: src/routes/routes.tsx (hunk 1 above)
Comment (fasbrothers) [reply src/routes/routes.tsx]: I realised that there is a load of full page with strangely. Therefore, I removed from lazy. I will test it again. If it works fine, I revert them.

File: src/types/auth.types.ts [hunk 2]
```diff
@@ -34,6 +93,10 @@ export interface PasswordInputProps<TFieldValues extends FieldValues> {
...
   placeholder: string;
   control: Control<TFieldValues>;
-  confirmPassword?: string;
-  setIsPasswordValid: React.Dispatch<React.SetStateAction<boolean>>;
+  errors: FieldErrors<TFieldValues>;
+}
+
+export interface CommonResponse {
+  success: boolean;
+  message: 'string';
```
Comment (MdRasulov) [inline src/types/auth.types.ts]: seems like you accidentaly put type `string` into brackets?

Comment (fasbrothers) [reply src/types/auth.types.ts]: Yes, I made accidentaly

**Review by Sinumi — APPROVED** [review]

**Review by MdRasulov — APPROVED** [review]

File: src/api/mutations/account/useUpdateEmailMutation.ts [hunk 3]
```diff
@@ -7,17 +7,15 @@ import { ENDPOINTS } from '@src/config/endpoints.config';
...
   const navigate = useNavigate();
 
   return useMutation({
     mutationFn: async () => {
-      const { data } = await httpClient.put<{ success: boolean; message: string }>(
-        ENDPOINTS.auth.updateEmail,
-        { email }
-      );
+      const { data } = await httpClient.put<CommonResponse>(ENDPOINTS.auth.updateEmail, { email });
       toast.success(data.message);
```
Comment (gitkoshelew) [inline src/api/mutations/account/useUpdateEmailMutation.ts]: shouldn't it be emitted only on success?

**Review by gitkoshelew — APPROVED** [review]

**Pull request merged**